csrc/multidevice/device_mesh.h

```diff
@@ -67,20 +85,42 @@ class DeviceMesh final {
     return -1;
   }
 
-  // Returns the device at a particular index in the mesh
+  // Returns the indices of a multi-dimensional mesh, or an empty vector 
+  // if device is not present
+  std::vector<int64_t> getLocalIndices(const DeviceIdxType device) const;
+
+  // Returns the device at a particular global index in the mesh
   DeviceIdxType at(int64_t index) const {
     return vector_.at(index);
   }
 
   bool operator==(const DeviceMesh& other) const {
-    return vector_ == other.vector();
+    return vector_ == other.vector() && shape_ == other.shape();
   }
 
+  // Returns the max device index in the DeviceMesh.
+  DeviceIdxType maxDeviceIdx() const;
+
+  // Returns devices of this device's team given which axis of the DeviceMesh
+  // Ex: [[0 1 2]
+  //      [3 4 5]]
+  // getTeam(4, 0) = {3, 4, 5}
+  // getTeam(4, 1) = {1, 4}
+  // TODO: calculate this once and save it
+  std::vector<DeviceIdxType> getTeam(DeviceIdxType device, int64_t axis = 0)
```

Comment:
also, we have been using the type `Team` to alias `std::vector<DeviceIdxType>`, I suggest we get rid of this alias -- otherwise we might want this function to use it as return type